app/src/components/Map/FilterPanel.jsx

```diff
@@ -66,7 +67,7 @@ class FilterPanel extends Component {
   }
 
   updateFilters(filter, index) {
-    const updatedFilters = _.cloneDeep(this.props.flags);
+    const updatedFilters = cloneDeep(this.props.flags);
```

Comment:
I'm not sure this is needed. I think that you can clone only the part of the tree that needs to be updated, and avoid re-rendering the rest. A shallow clone might do the trick.
That said, this might not be the case so we would need to do some testing.
[More info]
https://twitter.com/dan_abramov/status/688087202312491008?lang=es